Tests/UnitTests/Attribution/AttributionPosterTests.swift

```diff
@@ -76,43 +77,104 @@ class AttributionPosterTests: TestCase {
 
     func testPostAttributionDataSkipsIfAlreadySent() {
         let userID = "userID"
+
         attributionPoster.post(attributionData: ["something": "here"],
                                fromNetwork: .adjust,
                                networkUserId: userID)
+
+        expect(self.backend.invokedPostAdServicesTokenCount) == 0
         expect(self.subscriberAttributesManager.invokedConvertAttributionDataAndSetCount) == 1
 
         attributionPoster.post(attributionData: ["something": "else"],
                                fromNetwork: .adjust,
                                networkUserId: userID)
+        expect(self.backend.invokedPostAdServicesTokenCount) == 0
         expect(self.subscriberAttributesManager.invokedConvertAttributionDataAndSetCount) == 1
-
     }
 
     func testPostAttributionDataDoesntSkipIfNetworkChanged() {
         let userID = "userID"
-
+        backend.stubbedPostAdServicesTokenCompletionResult = Result.success(())
         attributionPoster.post(attributionData: ["something": "here"],
                                fromNetwork: .adjust,
                                networkUserId: userID)
+        expect(self.backend.invokedPostAdServicesTokenCount) == 0
         expect(self.subscriberAttributesManager.invokedConvertAttributionDataAndSetCount) == 1
 
         attributionPoster.post(attributionData: ["something": "else"],
                                fromNetwork: .facebook,
                                networkUserId: userID)
-
+        expect(self.backend.invokedPostAdServicesTokenCount) == 0
         expect(self.subscriberAttributesManager.invokedConvertAttributionDataAndSetCount) == 2
     }
 
     func testPostAttributionDataDoesntSkipIfDifferentUserIdButSameNetwork() {
+        backend.stubbedPostAdServicesTokenCompletionResult = Result.success(())
+
         attributionPoster.post(attributionData: ["something": "here"],
                                fromNetwork: .adjust,
                                networkUserId: "attributionUser1")
+        expect(self.backend.invokedPostAdServicesTokenCount) == 0
         expect(self.subscriberAttributesManager.invokedConvertAttributionDataAndSetCount) == 1
 
         attributionPoster.post(attributionData: ["something": "else"],
                                fromNetwork: .adjust,
                                networkUserId: "attributionUser2")
+        expect(self.backend.invokedPostAdServicesTokenCount) == 0
         expect(self.subscriberAttributesManager.invokedConvertAttributionDataAndSetCount) == 2
     }
 
+#if canImport(AdServices)
+    @available(iOS 14.3, macOS 11.1, macCatalyst 14.3, *)
+    func testPostAdServicesTokenIfNeededSkipsIfAlreadySent() throws {
+        guard #available(iOS 14.3, macOS 11.1, macCatalyst 14.3, *) else {
+            throw XCTSkip("Required API is not available for this test.")
+        }
```

Comment:
These 3 are duplicated in all of these tests.
Just a suggestion, feel free to ignore, but you could extract a `BaseAttributionPosterTests` that has the properties and `setUp` in this class that `AttributionPosterTests` and a new `AdServicesAttributionPosterTests` class can inherit.
Then you can put this inside of a shared `setUp` as well as making that whole class:
```swift
@available(iOS 14.3, macOS 11.1, macCatalyst 14.3, *)
```